integrations/scarf/src/scarf-script.raw.js

```diff
@@ -0,0 +1,57 @@
+(function () {
+    try {
+        const pixelID = '1234567890';
+        const GRANTED_COOKIE = '__gitbook_cookie_granted';
+
+        function getCookie(cname) {
+            const name = `${cname}=`;
+            const decodedCookie = decodeURIComponent(document.cookie);
+            const ca = decodedCookie.split(';');
+            for (let i = 0; i < ca.length; i++) {
+                let c = ca[i];
+                while (c.charAt(0) === ' ') {
+                    c = c.substring(1);
+                }
+                if (c.indexOf(name) === 0) {
+                    return c.substring(name.length, c.length);
+                }
+            }
+            return '';
+        }
+
+        let lastHref = null;
+
+        const cookieGranted = getCookie(GRANTED_COOKIE);
+
+        function sendScarfPing() {
+            if (cookieGranted !== 'yes') {
```

Comment:
Hey, sorry about the confusion earlier, I suggested the wrong change.

The correct version should just be:

```js
if (getCookie(GRANTED_COOKIE) !== 'yes') {
```

And there’s no need for the `cookieGranted` variable. 

The reason is: if we read the cookie value once at the top level, it won’t reflect any updates after the user accepts cookies I think. 

The integration would remain disabled even after consent. Reading it dynamically ensures it always checks the latest value.

We need to do the same exact thing in all other integrations - if possible.